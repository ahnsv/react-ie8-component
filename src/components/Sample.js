const React = require("react");
const Component = React.Component;
const styles = require("./Sample.css");
export default class Sample extends Component {
	render() {
		return <div className={styles.hi}>Sample Hi</div>;
	}
}
