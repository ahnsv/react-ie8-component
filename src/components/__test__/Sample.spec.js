import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Sample from '../Sample'
const configure = require("enzyme").configure;
const Adapter = require("enzyme-adapter-react-14");
configure({ adapter: new Adapter() });

describe('Sample', () => {
    it('should be defined', () => {
      expect(Sample).toBeDefined();
    });
    it('should render correctly', () => {
      const tree = shallow(
        <Sample name='Sample test' />
      );
      expect(tree).toMatchSnapshot();
    });
   });