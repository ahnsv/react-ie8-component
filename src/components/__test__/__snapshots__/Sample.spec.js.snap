// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sample
    name="Sample test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Sample Hi",
      "className": undefined,
    },
    "ref": null,
    "rendered": "Sample Hi",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sample Hi",
        "className": undefined,
      },
      "ref": null,
      "rendered": "Sample Hi",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
